client-v2/src/components/FrontsEdit/ArticleFragmentFormInline.tsx

```diff
@@ -215,9 +217,42 @@ class FormComponent extends React.Component<Props, FormComponentState> {
       imageCutoutReplace,
       cutoutImage,
       imageSlideshowReplace,
-      isBreaking
+      isBreaking,
+      editMode
     } = this.props;
 
+    // const assertion to prevent type widening, necessary to keep `type KickerField` below DRY
+    // see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions
+    const kickerFields = ['showKickerTag', 'showKickerSection'] as const;
+
+    // In an effort to be DRY, create a type of all values from the `kickerFields` array that can be accessed by a number.
+    // That is, a union of the values.
+    // i.e. KickerField = 'showKickerTag' | 'showKickerSection'
+    type KickerField = typeof kickerFields[number];
+
+    const setCustomKicker = (customKickerValue: string) => {
+      change('customKicker', customKickerValue);
+      change('showKickerCustom', true);
+      kickerFields.forEach(field => change(field, false));
+    };
+
+    const handleKickerChange = (
+      customKickerValue: string,
+      showField: boolean,
+      fieldName: KickerField
+    ) => {
+      if (editMode === 'editions') {
+        setCustomKicker(customKickerValue);
+      } else {
+        if (!showField) {
+          kickerFields.forEach(field => change(field, field === fieldName));
+          change('showKickerCustom', false);
+        } else {
+          change(fieldName, false);
+        }
```

Comment:
I'd switch the order of this if statement if it were me.

Reply:
The logic seems to be the reverse of what I would expect. Why `!showField` => `change(fieldName, false)` ???

Reply:
Ah - the method is a toggle and `showField` is the **old** value before the toggle. Can the `!` be moved to where handleKickerChange is called? Alternatively (at the very least) let's rename the params to make it clear that it's the `old` value.